processor/prometheusdiscoveryprocessor/prometheus_discovery_processor.go

```diff
@@ -50,48 +56,67 @@ func (pdp *prometheusDiscoveryProcessor) ProcessMetrics(_ context.Context, pdm p
 	rms := pdm.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		rm := rms.At(i)
-		attrs := rm.Resource().Attributes()
 
-		job, ok := attrs.Get("job")
-		if !ok {
-			// metric doesn't have 'job' label, ignore it.
-			continue
-		}
+		attrs := rm.Resource().Attributes()
 
-		instance, ok := attrs.Get("instance")
-		if !ok {
-			// metric doesn't have 'instance' label, ignore it.
-			continue
-		}
+		if key, ok := getCacheKeyForResource(rm.Resource()); ok {
 
-		key := cacheKey{
-			job:      job.StringVal(),
-			instance: instance.StringVal(),
+			if source, ok := attrs.Get("source"); ok && source.StringVal() == "prometheus_discovery" {
+				// attrs came from prometheus_discovery; cache them
+				pdp.attributeCache[key] = attrs
+			} else {
+				// these are "normal" metrics, enrich the resource with cached attrs
+				pdp.enrichResource(rm.Resource())
+			}
 		}
 
-		ilms := rm.InstrumentationLibraryMetrics()
-		for j := 0; j < ilms.Len(); j++ {
-			ms := ilms.At(j).Metrics()
-			for k := 0; k < ms.Len(); k++ {
-				met := ms.At(k)
+		// I don't think we need to go this deep given that we are just dealing with resources, for now
+		/*
+			ilms := rm.InstrumentationLibraryMetrics()
+			for j := 0; j < ilms.Len(); j++ {
+				ms := ilms.At(j).Metrics()
+				for k := 0; k < ms.Len(); k++ {
+					met := ms.At(k)
+					if met.Name() == "present" {
 
-				if met.Name() == "present" {
-					if source, ok := attrs.Get("source"); ok && source.StringVal() == "prometheus_discovery" {
-						pdp.attributeCache[key] = attrs
```

Comment:
This line here is caching the resource attributes if there is a present metric. If the resource has the special attribute: `source: prometheus_discovery` do we still need to search for a present metric, or can we assume that the attributes should be cached due to the existence of that label alone? In this latest set of changes, I assumed that was the case. There's also a chance that I'm horrible mistaken, so please correct me if I am.